### PR TITLE
refactor: change the key to a pair of number

### PR DESCRIPTION
**Pull request by leanmendoza**

### CODE REVIEW

File: test/utils/index.ts [hunk 1]
```diff
@@ -47,3 +64,18 @@ export function shuffle<T = unknown>(value: T[]) {
     .sort((a, b) => a.sort - b.sort)
     .map(({ value }) => value)
 }
+
+export function parseStateString<T>(stateStr: string) {
+  const state = JSON.parse(stateStr)
+  const newState: State<T> = new Map()
+  for (const [key, value] of Object.entries(state)) {
+    const keys = key.split('.')
```
[the remark this answers is not in the document]
Comment (leanmendoza) [reply test/utils/index.ts:72]: changing it

File: src/index.ts [hunk 2]
```diff
@@ -130,12 +157,28 @@ export function crdtProtocol<
...
-    return { ...state } as State<T>
+    return state
+  }
+
+  /**
+   * Returns the current state
+   * @public
+   */
+  function cloneState(): State<T> {
+    const newState: State<T> = new Map()
```
[the remark this answers is not in the document]
Comment (leanmendoza) [reply src/index.ts:168]: We don't need to keep it, but that wouldn't work. As the values are other maps, they are copied by reference.

File: test/utils/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (leanmendoza) [reply test/utils/index.ts:72]: anyway, it's necessary to implement the parse and serialize (or stringify and parse)

File: test/utils/snapshot.ts [hunk 3]
```diff
@@ -72,26 +73,29 @@ export function snapshotTest<T = unknown>() {
...
   }
 
-  function parseState(state: ReturnType<CRDT<T>['getState']>): string {
-    const newState: State<string> = {}
-    for (const key in state) {
-      newState[key] = { ...state[key], data: parseData(state[key].data) }
+  function stateToString<T>(state: State<T>): string {
+    const objState = {}
+    for (const [key1, key2, value] of stateIterator(state)) {
+      objState[`${key1}.${key2}`] = {
```
[the remark this answers is not in the document]
Comment (leanmendoza) [reply test/utils/snapshot.ts:90]: to compare the saved string with the on-memory map

File: test/utils/index.ts [hunk 4]
```diff
@@ -47,3 +64,18 @@ export function shuffle<T = unknown>(value: T[]) {
...
+    const [key1, key2] = [parseInt(keys[0]), parseInt(keys[1])]
+    if (!newState.has(key1)) {
+      newState.set(key1, new Map())
+    }
+
+    newState.get(key1)!.set(key2, value !== null ? { ...(value as any) } : null)
```
[the remark this answers is not in the document]
Comment (leanmendoza) [reply test/utils/index.ts:78]: value is unknown, I need to cast it as something before accessing it

File: test/utils/index.ts [hunk 5]
```diff
@@ -16,17 +16,34 @@ export function compareStatePayloads<T = Buffer>(states: State<T>[]) {
...
-          s[key].timestamp === baseState[key].timestamp &&
-          compareData(s[key].data, baseState[key].data)
+  const key1EqualSizes = states.every((s) => s.size === baseState.size)
+  if (!key1EqualSizes) {
+    return false
+  }
+
+  return states.every((currentState) => {
+    for (const [key1, key2, baseStatePayload] of stateIterator(baseState)) {
+      const key2EqualSizes = states.every(
```
[the remark this answers is not in the document]
Comment (leanmendoza) [reply test/utils/index.ts:28]: got it, I didn't want to change the original code too much

File: src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (leanmendoza) [reply src/index.ts]: deleted it

File: test/utils/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (leanmendoza) [reply test/utils/index.ts]: `stateToString` and `stateFromString` were implemented as helpers function in `test/utils`

File: test/utils/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (leanmendoza) [reply test/utils/index.ts]: done

**Review by gonpombo8 — APPROVED** [review]

**Pull request merged**